### DISCUSSION TITLE
How to hide a specific window and not all window of the application?

### DISCUSSION THREAD

**Discussion by kakawait** [Hammerspoon/hammerspoon#3202] [Q&A] [answered]
I'd like to hide only one or several windows but not the whole app.
Do you think is possible? As of today I was only capable to hide every windows of an app.

**Comment by latenitefilms** [comment, marked as answer]
It's definitely possible, but you'll need to use the Accessibility Framework.

**Comment by benjaminwelch** [comment]
I'm a beginner and couldn't figure out how to do this with the Accessibility Framework but I came up with a solution that works for my needs. Posting in case anyone else wants to try it.

I discovered that you can move individual windows to another Space on the Mac without sending the whole application. This in effect hides the application if you send it to a "empty" space. And with another command, you can bring back all the windows you sent to that Space and focus the most recent window you sent.

Here is the code I used:
```lua
-- move the currently focused window to Space with ID 2623
hs.hotkey.bind("cmd", "ctrl", "h", function()
  local win = hs.window.focusedWindow()
  local space = hs.spaces.focusedSpace()
  -- hs.alert(space)
  hs.spaces.moveWindowToSpace(win, 2623)
end)

-- Bring all "hidden windows" back to main Space
hs.hotkey.bind("cmd", "ctrl", "alt", "h", function()
  local wins = hs.spaces.windowsForSpace(2623)
  if #wins > 0 then
    local focused = hs.window.find(wins[-1])
  end
  for key in pairs(wins) do
    hs.spaces.moveWindowToSpace(wins[key], 1)
  end
  -- hs.alert(focused)
  hs.window.find(focused):focus()
end)
```

To adapt this script for your needs:
`local win = hs.window.focusedWindow()`
`local space = hs.spaces.focusedSpace()`
These two find the current Space and focused Window.

Uncomment this line and hit the hotkey to find the ID's for your Spaces. First check the ID of your main Space, write that down. For me it's `1`. Then switch to the Space you'll be using to hide the windows and hit the hotkey again. Write down the ID of that Space (`2623`) for me.
 ` -- hs.alert(space)`

Edit this line and replace `2623` with the ID of the Space for hiding windows:
`hs.spaces.moveWindowToSpace(win, 2623)`

In the second hotkey binding, replace `2623` with the same ID:
`local wins = hs.spaces.windowsForSpace(2623)`

In this line, change the `1` to be the ID of your main Space:
`hs.spaces.moveWindowToSpace(wins[key], 1)`

And this line is focusing the windows you just "unhid":
`  hs.window.find(focused):focus()`

I hope this is helpful. Bit of a workaround but works for my needs. Let me know if you try this out and it works for you.
